lib/common/http2client.c

```diff
@@ -312,6 +333,11 @@ static int on_head(struct st_h2o_http2client_conn_t *conn, struct st_h2o_http2cl
         return 0;
     }
 
+    if (extract_content_length(&stream->input.headers, &stream->input.remaining_content_length, err_desc) != 0) {
```

Comment:
FWIW, honestly I think that content-length should not be passed as a direct member of `h2o_httpclient_on_head_t` of type `size_t`, rather than as one of the header fields carried by `h2o_httpclient_on_head_t::headers`. We do not want each module (e.g., http1client.c, http2client.c, lib/core/proxy.c) to parse content-length.

That said, changing the design is going to be a bigger surgery and I'm fine with the proposed change as-is (we can refactor at a later moment).

Reply:
Yes, I also think that think that that refactoring is outside of the scope of this PR.